src/components/icon/icon.spec.js

```diff
@@ -514,29 +510,39 @@ describe('MdIcon service', function() {
     });
 
     describe('icon group is not found', function() {
-      it('should log Error', function() {
-        var msg;
-        try {
-          $mdIcon('notfound:someIcon')
-            .catch(function(error) {
-              msg = error.data;
-            });
+      it('should log Error and reject', inject(function($log) {
+        var errorMessage = 'Cannot GET notfoundgroup.svg';
+        var caughtRejection = false;
+
+        $mdIcon('notfound:someIcon')
+          .catch(function(error) {
+            expect(error.data).toBe(errorMessage);
+            caughtRejection = true;
+          });
 
-          $httpBackend.flush();
-        } finally {
-          expect(msg).toEqual('Cannot GET notfoundgroup.svg');
-        }
-      });
+        $httpBackend.flush();
+
+        expect(caughtRejection).toBe(true);
+        expect($log.warn.logs[0]).toEqual([errorMessage]);
+      }));
     });
 
     describe('icon is not found', function() {
-      it('should not throw Error', function() {
-        expect(function(){
-          $mdIcon('notfound');
+      it('should log Error and reject', inject(function($log) {
```

Comment:
tl;dr: Yes

Longer answer:

**Before this PR**
- pre-1.6:
  - `$mdIcon('notfound')`: Nothing happens
  - `$mdIcon('notfound').catch(callback)`: `callback` gets called
- post-1.6:
  - `$mdIcon('notfound')`: 2 PURs (thrown in tests, logged in production)
  - `$mdIcon('notfound').catch(callback)`: `callback` gets called + 1 PUR (thrown in tests, logged in production)

**After this PR**
- pre-1.6:
  - `$mdIcon('notfound')`: Nothing happens
  - `$mdIcon('notfound').catch(callback)`: `callback` gets called
- post-1.6:
  - `$mdIcon('notfound')`: 1 PUR (thrown in tests, logged in production)
  - `$mdIcon('notfound').catch(callback)`: `callback` gets called